spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1812,8 +1812,8 @@ def _do_run(self, n_machine_time_steps, loading_done, run_until_complete):
         if self._use_virtual_board:
             logger.warn(
                 "Application will not actually be run as on a virtual board")
-        elif self._executable_start_type == \
-                ExecutableStartType.NO_APPLICATION:
+        elif (len(self._executable_types) == 0 and
+                ExecutableType.NO_APPLICATION in self._executable_types):
```

Comment:
These can't both be true...

Reply:
doh. fixed. need to run these unit tets more often. lol

Reply:
fixed